client/ionic/src/components/ShareButton.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { useState } from "react";
+import { IonContent, IonButton } from "@ionic/react";
+import { SocialSharing } from '@ionic-native/social-sharing';
+
+
+interface ShareConfig {
+    shareSTR: string;
+  }
+
+const ShareButton: React.FC<ShareConfig> = ({ shareSTR }) => {
+    function share(){
+        SocialSharing.share(shareSTR)
+    }
+    return (
+        <IonContent>
+             <IonButton
+              onClick={share}
+              expand="block"
+            >
```

Comment:
another nitpicky one here, but please just do a quick check of your indentation across the file. The Prettier / Editorconfig changes in progress will help in the future